File: sharp_brain/sharp_brain/Neural/OutputNeuron.cs
using SharpBrain.Activations;

namespace SharpBrain.Neural
{
    public class OutputNeuron : WorkingNeuron
    {
        public OutputNeuron(int layerIndex, int index, Activation activation)
            : base(layerIndex, index, activation)
        {

        }

        public void CalculateDelta(double expected)
        {
            var derivative = activation.Derivative(netInput);
            var difference = expected - Value;
            delta = derivative * difference;
        }
    }
}

File: kotlin_brians/src/main/kotlin/neurals/OutputNeuron.kt
package neurals

import activations.Activation

class OutputNeuron(layerIndex: Int, index: Int, activation: Activation) : WorkingNeuron(layerIndex, index, activation) {
    fun calculateDelta(expectedValue: Double) {
        val difference: Double = expectedValue - value
        val derivative: Double = activation.derivative(netInput)
        delta = difference * derivative
    }
}